Lambdas/GHOAuth/OAuthLambda.swift

```diff
@@ -0,0 +1,171 @@
+import AsyncHTTPClient
+import AWSLambdaRuntime
+import AWSLambdaEvents
+import DiscordBM
+import Foundation
+import SotoSecretsManager
+import GHHooksLambda
+import Models
+import JWTKit
+import SharedServices
+
+@main
+struct GHOAuthHandler: LambdaHandler {
+    typealias Event = APIGatewayV2Request
+    typealias Output = APIGatewayV2Response
+
+    let client: HTTPClient
+    let logger: Logger
+    let secretsRetriever: SecretsRetriever
+    let discordClient: any DiscordClient
+    let jsonDecoder: JSONDecoder
+    let jsonEncoder: JSONEncoder
+    let userService: UserService
+
+    init(context: LambdaInitializationContext) async throws {
+        self.client = HTTPClient(eventLoopGroupProvider: .shared(context.eventLoop))
+        self.logger = context.logger
+
+        let awsClient = AWSClient(httpClientProvider: .shared(client))
+        self.secretsRetriever = SecretsRetriever(awsClient: awsClient, logger: logger)
+
+        let botToken = try await secretsRetriever.getSecret(arnEnvVarKey: "BOT_TOKEN_ARN")
+        self.discordClient = await DefaultDiscordClient(httpClient: client, token: botToken)
+
+        self.userService = UserService(awsClient, logger)
+
+        self.jsonDecoder = JSONDecoder()
+        self.jsonEncoder = JSONEncoder()
+    }
+
+    func handle(_ event: APIGatewayV2Request, context: LambdaContext) async -> APIGatewayV2Response {
+        logger.trace("Received event: \(event)")
+
+        guard let code = event.queryStringParameters?["code"] else {
+            return .init(statusCode: .badRequest, body: "Missing code query parameter")
+        }
+
+        let accessToken: String
+
+        do {
+            accessToken = try await getGHAccessToken(code: code)
+        } catch {
+            logger.error("Error getting access token: \(error)")
+            return .init(statusCode: .badRequest, body: "Error getting access token")
+        }
+
+        let userID: Int
+
+        do {
+            userID = try await getGHUserID(accessToken: accessToken)
+        } catch {
+            logger.error("Error getting user ID: \(error)")
+            return .init(statusCode: .badRequest, body: "Error getting user")
+        }
+
+        let jwt: GHOAuthPayload
+
+        do {
+            jwt = try await verifyState(state: String(event.queryStringParameters?["state"] ?? ""))
+        } catch {
+            logger.error("Error during state verification: error: \(error)", metadata: [
+                "state": .string(event.queryStringParameters?["state"] ?? "")
+            ])
+            return .init(statusCode: .badRequest, body: "Error verifying state")
+        }
+
+        do {
+            try await userService.linkGithubID(to: jwt.discordID.rawValue, githubID: "\(userID)")
+        } catch {
+            logger.error("Error linking user with Discord ID \(jwt.discordID) and GitHub ID \(userID): \(error)")
+            return .init(statusCode: .badRequest, body: "Error linking user")
+        }
+
```

Comment:
Here, also update the Discord message using the `interactionToken` in the JWT:
```suggestion

        do {
            try await discordClient.updateOriginalInteractionResponse(
                token: jwt.interactionToken,
                payload: .init(
                    embeds: [.init(
                        description: """
                    Successfully linked your Discord account to GitHub account with ID: \(userID)
                    """,
                        color: .green
                    )]
                )
            ).guardSuccess()
        } catch {
            logger.warning("Received Discord error while updating interaction", metadata: [
                "error": "\(error)"
            ])
        }

```

Reply:
Bonus points if you bother to change the `User` struct to also contain `let login: String`, which is basically the name of a GitHub account. Then mention the `login` in the Discord message, like `Successfully linked your Discord account to GitHub account with name: \(user.login)`.

You'll need to change the `getGHUserID` func to `getGHUser` and make it return the whole `User` struct so you can access the `user.login` for the Discord message.